test/connection_test.rb

```diff
@@ -326,126 +319,71 @@ def test_proxy_accepts_hash
   end
 
   def test_proxy_accepts_http_env
-    with_env 'http_proxy' => "http://duncan.proxy.com:80" do
+    with_env 'http_proxy', "http://duncan.proxy.com:80" do
       conn = Faraday::Connection.new
       assert_equal 'duncan.proxy.com', conn.proxy.host
     end
   end
 
   def test_proxy_accepts_http_env_with_auth
-    with_env 'http_proxy' => "http://a%40b:my%20pass@duncan.proxy.com:80" do
+    with_env 'http_proxy', "http://a%40b:my%20pass@duncan.proxy.com:80" do
       conn = Faraday::Connection.new
       assert_equal 'a@b',     conn.proxy.user
       assert_equal 'my pass', conn.proxy.password
     end
   end
 
   def test_proxy_accepts_env_without_scheme
-    with_env 'http_proxy' => "localhost:8888" do
+    with_env 'http_proxy', "localhost:8888" do
       uri = Faraday::Connection.new.proxy[:uri]
       assert_equal 'localhost', uri.host
       assert_equal 8888, uri.port
     end
   end
 
   def test_no_proxy_from_env
-    with_env 'http_proxy' => nil do
+    with_env 'http_proxy', nil do
       conn = Faraday::Connection.new
       assert_nil conn.proxy
     end
   end
 
   def test_no_proxy_from_blank_env
-    with_env 'http_proxy' => '' do
+    with_env 'http_proxy', '' do
       conn = Faraday::Connection.new
       assert_nil conn.proxy
     end
   end
 
   def test_proxy_doesnt_accept_uppercase_env
-    with_env 'HTTP_PROXY' => "http://localhost:8888/" do
+    with_env 'HTTP_PROXY', "http://localhost:8888/" do
       conn = Faraday::Connection.new
       assert_nil conn.proxy
     end
   end
 
-  def test_proxy_requires_uri
-    conn = Faraday::Connection.new
-    assert_raises ArgumentError do
-      conn.proxy :uri => :bad_uri, :user => 'rick'
-    end
-  end
-
-  def test_proxy_allowed_when_url_in_no_proxy_list
-    with_env 'http_proxy' => 'http://proxy.com', 'no_proxy' => 'example.com' do
-      conn = Faraday::Connection.new
-      assert_equal conn.proxy_allowed?('http://example.com'), false
-    end
-  end
```

Comment:
Please reintroduce all the deleted test as they're covering some important edge-cases.
This can be done without using the `proxy_allowed?` method:

- If the test asserts `assert_equal conn.proxy_allowed?('http://example.com'), false`, then you can change it into `assert_nil conn.proxy`
- If the test asserts `assert_equal conn.proxy_allowed?('http://prefixedexample.com'), true` than you can change it into `assert_equal 'proxy.com', conn.proxy.host`